src/pkg/build/conveyorPacker_debootstrap.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package build
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"runtime"
+)
+
+//"github.com/singularityware/singularity/src/pkg/image"
+
+// DebootstrapConveyor holds stuff that needs to be packed into the bundle
+type DebootstrapConveyor struct {
+	recipe Definition
+	tmpfs  string
+}
+
+// DebootstrapConveyorPacker only needs to hold the conveyor to have the needed data to pack
+type DebootstrapConveyorPacker struct {
+	DebootstrapConveyor
+}
+
+// Get downloads container information from the specified source
+func (c *DebootstrapConveyor) Get(recipe Definition) (err error) {
+
+	c.recipe = recipe
+
+	//get mirrorURL and OSVerison components to definition
+	OSVersion, ok := recipe.Header["MirrorURL"]
+	if ok != false {
+		return fmt.Errorf("Invalid debootstrap header, no MirrorURL specified")
+	}
+
+	MirrorURL, ok := recipe.Header["OSVersion"]
+	if ok != false {
+		return fmt.Errorf("Invalid debootstrap header, no OSVersion specified")
+	}
+
+	//check for debootstrap on system(script using "singularity_which" not sure about its importance)
+	debootstrapPath, err := exec.LookPath("debootstrap")
+	if err != nil {
+		return fmt.Errorf("debootstrap is not in PATH... Perhaps 'apt-get install' it?%v", debootstrapPath)
+	}
+
+	c.tmpfs, err = ioutil.TempDir("", "temp-debootstrap-")
+	if err != nil {
+		return
+	}
+
+	//Dont know what this does...
+	//REQUIRES=`echo "${INCLUDE:-}" | sed -e 's/\s/,/g'`
+	requires := ""
+
+	//run debootstrap command
+	//$DEBOOTSTRAP_PATH --variant=minbase --exclude=openssl,udev,debconf-i18n,e2fsprogs --include=apt,$REQUIRES --arch=$ARCH '$OSVERSION' '$SINGULARITY_ROOTFS' '$MIRRORURL'
+	cmd := exec.Command(debootstrapPath, `--variant=minbase --exclude=openssl,udev,debconf-i18n,e2fsprogs`, `--include=apt,`+requires, `--arch=`+runtime.GOARCH, OSVersion, c.tmpfs, MirrorURL)
```

Comment:
What is the output of `runtime.GOARCH` ?? Is it what Debian expects? i.e. the `amd64`, `arm64`, etc...

Reply:
I think so. Check out the list under "optional environment variables"
https://golang.org/doc/install/source#environment

Reply:
It outputs in that format and a list of the potential values can be found here https://github.com/golang/go/blob/master/src/go/build/syslist.go